src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -288,6 +292,28 @@ public BaseDataObject(final byte[] newData, final String name, final String form
         }
     }
 
+    public BaseDataObject(final byte[] newData, final String name, @Nullable final String form, IBaseDataObject tld) {
+        setData(newData);
```

Comment:
Any reason not to use `this(newData, name, form);`? Do you not want the creation timestamp set??